File: backend/txn-service/src/main/java/com/meiyun/txn/Arrival.java
package com.meiyun.txn;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.OffsetDateTime;

/**
 * 到店登记（接待台/候诊队列的一行）。
 *
 * <p>来源：WALK_IN（自然到店）/ REFERRAL（转介绍）/ MARKETING（线上营销渠道）/ APPOINTMENT（预约签到自动生成）。
 *
 * <p>状态机对齐前端 stores/arrival.ts 活规格：
 * <pre>
 *   WAITING（候诊中）→ TRIAGED（已分诊·入位）→ CALLED（已叫号）→ DONE（已完成）
 *   WAITING → LEFT（客户离开）
 *   CALLED → TRIAGED（叫号后退回重分诊，后端预留，本批页面不暴露按钮）
 * </pre>
 * queue_no 为门店当日递增排队号（与 AH 单号分离）；appt_no 为预约来源幂等键（先查后插）。
 */
@Entity
@Table(name = "arrival", indexes = {
        @Index(name = "idx_arrival_store_arrived", columnList = "store_code,arrived_at")
})
@Getter @Setter @NoArgsConstructor
public class Arrival {

    @Id
    @Column(name = "ah_no", length = 24)
    private String ahNo;

    /** 状态：WAITING / TRIAGED / CALLED / DONE / LEFT。 */
    @Column(nullable = false, length = 16)
    private String status;

    @Column(name = "customer_id", nullable = false, length = 16)
    private String customerId;

    @Column(name = "store_code", nullable = false, length = 16)
    private String storeCode;

    /** 到店渠道：WALK_IN / REFERRAL / MARKETING / APPOINTMENT。 */
    @Column(nullable = false, length = 16)
    private String channel;

    /** 门店当日排队号（max+1，与 AH 单号分离）。 */
    @Column(name = "queue_no", nullable = false)
    private Integer queueNo;

    @Column(length = 256)
    private String note;

    /** 关联预约号（APPOINTMENT 来源非空，幂等键）。 */
    @Column(name = "appt_no", length = 24)
    private String apptNo;

    @Column(name = "arrived_at", nullable = false)
    private OffsetDateTime arrivedAt;

    @Column(name = "called_at")
    private OffsetDateTime calledAt;

    @Column(name = "done_at")
    private OffsetDateTime doneAt;

    /** 离开/号源释放时间（WAITING → LEFT；手工释放与超时自动释放同列，原因见审计 payload）。 */
    @Column(name = "left_at")
    private OffsetDateTime leftAt;

    /** 登记/操作人工号（JWT 登录人，不信入参）。 */
    @Column(nullable = false, length = 32)
    private String operator;

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;

    @PrePersist
    void prePersist() {
        if (createdAt == null) createdAt = OffsetDateTime.now();
        if (arrivedAt == null) arrivedAt = OffsetDateTime.now();
        if (status == null) status = "WAITING";
    }
}

File: frontend/src/api/arrival.ts
// ============================================================
// 接待台 / 候诊分诊 API（对接 txn-service 到店域：/api/txn/arrivals）
//
// 到店登记 → 候诊（WAITING）→ 分诊入位（TRIAGED，同事务建方案草稿）
//   → 叫号（CALLED）→ 完成（DONE）；分诊单可改派（reassign）。
// 状态/类型码对齐后端英文枚举，中文化由页面模板现有 chip/字典完成。
// ============================================================
import client from './client'

export type ArrivalStatus = 'WAITING' | 'TRIAGED' | 'CALLED' | 'DONE' | 'LEFT'
export type TriageType = 'CONSULT' | 'MEDICAL' | 'SERVICE'
export type ArrivalChannel = 'WALK_IN' | 'REFERRAL' | 'MARKETING' | 'APPOINTMENT'

/** 内联分诊单读模型：assignedToName 为首诊负责人名，ownerName 为改派后当前负责人名。 */
export interface TriageViewDTO {
  id: string
  arrivalId: string
  customerId: string
  type: TriageType | string
  assignedTo: string
  assignedToName: string | null
  forwardedTo: string | null
  forwardedToName: string | null
  ownerName: string | null
  note: string | null
  planId: string | null
  editedBy: string | null
  editedAt: string | null
}

/** 到店记录读模型（后端富化客户名/掩码手机号 + 内联当前分诊单）。 */
export interface ArrivalViewDTO {
  id: string
  ahNo: string
  customerId: string
  storeCode: string
  customerName: string
  phoneMask: string
  channel: ArrivalChannel | string
  queueNo: number
  status: ArrivalStatus | string
  note: string | null
  apptNo: string | null
  arrivedAt: string
  calledAt: string | null
  doneAt: string | null
  triage: TriageViewDTO | null
}

export interface CheckInCmd {
  customerId: string
  channel?: string
  note?: string
}

export interface TriageCmd {
  type: string
  assignedTo: string
  note?: string
}

export interface ReassignCmd {
  newAssignedTo: string
}

/** 今日队列：默认今日，支持日期/门店/状态过滤；接待台与候诊看板共用一次拉全。 */
export const listArrivals = (params?: { date?: string; storeCode?: string; status?: string }) =>
  client.get<ArrivalViewDTO[]>('/txn/arrivals', { params })

/** 前台手工到店登记（门店由后端按 JWT 当前本店落，body 不传 storeCode）。 */
export const checkInArrival = (cmd: CheckInCmd) =>
  client.post<ArrivalViewDTO>('/txn/arrivals', cmd)

/** 分诊入位：WAITING → TRIAGED，同事务建 consult_plan 空草稿（planId 在 triage.planId 回查）。 */
export const triageArrival = (ahNo: string, cmd: TriageCmd) =>
  client.post<ArrivalViewDTO>(`/txn/arrivals/${ahNo}/triage`, cmd)

/** 改派：仅更新当前分诊单 forwardedTo，不动方案草稿归属。 */
export const reassignArrival = (ahNo: string, cmd: ReassignCmd) =>
  client.post<ArrivalViewDTO>(`/txn/arrivals/${ahNo}/reassign`, cmd)

/** 叫号：TRIAGED → CALLED（幂等）。 */
export const callArrival = (ahNo: string) =>
  client.post<ArrivalViewDTO>(`/txn/arrivals/${ahNo}/call`, {})

/** 完成接诊：TRIAGED/CALLED → DONE（幂等）。 */
export const doneArrival = (ahNo: string) =>
  client.post<ArrivalViewDTO>(`/txn/arrivals/${ahNo}/done`, {})
